File: Task-03/Ruby/Prime number.rb
print "Enter a value for n: "
n = gets.chomp.to_i

if n <= 1
  puts "#{n} is not a prime number"
else
  (2..n).each do |i|
    flag = 0
    (2..Math.sqrt(i).to_i).each do |j|
      if (i % j).zero?
        flag = 1
        break
      end
    end
    puts "#{i} is a prime number" if flag.zero?
  end
end

File: Task-03/Go/Prime number.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Print("Enter a value for n: ")

	if n <= 1 {
		fmt.Printf("%d is not a prime number\n", n)
	} else {
		for i := 2; i <= n; i++ {
			flag := 0
			for j := 2; j <= int(math.Sqrt(float64(i))); j++ {
				if i%j == 0 {
					flag = 1
					break
				}
			}
			if flag == 0 {
				fmt.Printf("%d is a prime number\n", i)
			}
		}
	}
}
